tests/tests/test_demo_artifact.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+# Copyright 2019 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        https://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from MenderAPI import authentication, deployments, auth_v2
+from mendertesting import MenderTesting
+import subprocess
+import requests
+import signal
+import pytest
+import logging
+import conftest
+import os
+
+# NOTE - The password is set on a per test-basis, as it is generated on the fly by
+# the demo script.
+auth = authentication.Authentication(
+    username='mender-demo', email='mender-demo@example.com')
+authv2 = auth_v2.auth = auth
+deploy = deployments.Deployments(auth, authv2)
+
+
+# @pytest.mark.incremental
+class TestDemoArtifact(MenderTesting):
+    """A simple class for testing the demo-Artifact upload."""
+
+    @pytest.fixture(scope="function")
+    def run_demo_script(self):
+        """Simple fixture which runs and brings down the demo script"""
+        # Run the demo script.
+        test_env = os.environ.copy()
+        test_env[
+            'DOCKER_COMPOSE_PROJECT_NAME'] = conftest.docker_compose_instance
+        proc = subprocess.Popen(
+            [
+                './demo', '--client', '-p', conftest.docker_compose_instance,
+                'up'
+            ],
+            cwd="..",
+            stdout=subprocess.PIPE,
+            env=test_env)
+        logging.info('Started the demo script')
+        password = ""
+        for line in iter(proc.stdout.readline, ''):
+            if "Login password:" in line.strip():
+                password = line[-13:-1]
+                break
+
+        logging.info('The login password:')
+        logging.info(password)
+        assert len(password) == 12
+
+        auth.password = password
+
+        yield proc
```

Comment:
After you have pushed the `addfinalizer` you can get rid of the rest of this function. `addfinalizer` is generally preferred over `yield`, since it will clean up even if the fixture crashes midway through, `yield` won't.